datageneration/generate_combination_table.py

```diff
@@ -146,27 +168,31 @@ def run(self, num_queries, number_of_entities_in_prompt, max_number_of_props_in_
         (2) within radius: a single radius within which all objects are located, (3) in area: general search for all objects
         within given area.
 
-        :param float area_chance: probability for picking up real name
-        :param str tag_list_path: Path to the CSV file containing all tags + a lot of meta info
-        :param str arbitrary_value_list_path: Path to CSV file containing samples for arbitrary and categorical values
-        :param str output_filename: Name under which the resulting output file should be stored (minus version specification)
-        :param str version: Defines whether "train", "dev", or "test" set is currently generated
-        :param bool save_json: Boolean that determines whether a JSON file should be saved or not
+        TODO: Write/update all docstrings, maybe use this text somewhere else
+
+        :param num_queries: (int) TODO
+        :param max_number_of_entities_in_prompt: (int) TODO
+        :param max_number_of_props_in_entity: (int) TODO
+        :param percentage_of_entities_with_props: (int) TODO
+        :param percentage_of_entities_with_props: (float) TODO
+        :return: loc_points (List[LocPoint])
         '''
         # ipek - node types are not used
         node_types = ["nwr", "cluster", "group"]
         loc_points = []
         for _ in tqdm(range(num_queries), total=num_queries):
             new_loc_points = []
-            entities = self.generate_entities(number_of_entities_in_prompt=number_of_entities_in_prompt,
-                                              max_number_of_props_in_entity=max_number_of_props_in_entity)
+            entities = self.generate_entities(max_number_of_entities_in_prompt=max_number_of_entities_in_prompt,
+                                              max_number_of_props_in_entity=max_number_of_props_in_entity,
+                                              percentage_of_entities_with_props=percentage_of_entities_with_props)
             area = self.generate_area()
             relations = self.generate_relations(num_entities=len(entities))
             new_loc_points.append(LocPoint(area=area, entities=entities, relations=relations).dict())
 
             # Clean up the output and remove all "None" added by the data model (the optional fields)
             for loc_point in new_loc_points:
                 for entity in loc_point["entities"]:
+
```

Comment:
We should retrieve the properties of pydantic objects as this loc_point.entities instead of loc_point['entities']. If you can not something is wrong, and the data model is treated as dictionary.